### ISSUE TITLE
[Bug] Creates a table when no newline after dash header

### ISSUE THREAD

**Issue by gbraad** [Zettlr/Zettlr#2461]
### Description

When I use a header demarker with dashes, it will create a table instead and removes actual content

### Reproducing

This is produced as follows:

```md
Rules
-----
There are only two real rules with Personal Kanban:

  1. Visualize your work
  2. Limit your work-in-progress
```

The following is created:

![image](https://user-images.githubusercontent.com/1894/133094491-b7453542-87f6-420f-bf95-1566d3e37c59.png)

This left the file modified as:

```patch
-Rules
------
-There are only two real rules with Personal Kanban:
+Rules
+-------
+There
```

### Zettlr Version

Stable (most recent version)

### Your Platform

- [X] Windows
- [X] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Not relevant for the issue

**Comment by gbraad** [comment]
Consistently reproducible

**Comment by nathanlesage** [comment]
Yup, because you must always separate a header from other block elements with a newline. That is pretty much to be expected, since there is no way the heuristics in the app are able to distinguish a Setext-heading from a table if you forget this.

**Comment by gbraad** [comment]
It is an issue with markdown parsers. Previously gitbook (pandoc and github) do not care about this case, and do what is expected... hence I never noticed the issue.

You wanna mark this as invalid?

**Comment by gbraad** [comment]
... but part of the issue is the fact that the content is truncated. If it just created a table, that would have been fine... but it creates a table and truncates.

**Comment by gbraad** [comment]
The issue exists in the beta and truncates to 4 characters

![image](https://user-images.githubusercontent.com/1894/133307767-b22ec6b1-71aa-4562-bdcf-9f4ec143aadc.png)

As mentioned, I understand the issue and that tables aren't part of the core Markdown spec, but the truncation seems to surely have a 'regression' as it is now even less characters.

**Comment by nathanlesage** [comment]
> but the truncation seems to surely have a 'regression' as it is now even less characters.

Yes, absolutely.

**Comment by nathanlesage** [comment]
This has fortunately been already fixed in 2.1.2
